### PR TITLE
Nonzero op

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

File: src/onnx/onnx.cpp [hunk 1]
```diff
@@ -2320,6 +2322,62 @@ struct onnx_parser
...
+        }
+
+        return indices;
+    }
+
+    instruction_ref
+    parse_nonzero(const std::string&, const node_info&, std::vector<instruction_ref> args)
+    {
+        migraphx::argument data_arg = args.back()->eval();
+        check_arg_empty(data_arg, "PARSE_NONZERO: cannot non-constant input!");
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:2355]:
```suggestion
        check_arg_empty(data_arg, "PARSE_NONZERO: cannot support non-constant input!");
```

Comment (scxiao) [reply src/onnx/onnx.cpp]: Thanks, fixed.

File: src/onnx/onnx.cpp [hunk 2]
```diff
@@ -2320,6 +2322,62 @@ struct onnx_parser
         MIGRAPHX_THROW("PARSE_ATEN: unsupported custom operator");
     }
 
+    template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
+    std::vector<std::size_t> nonzero_indices(std::vector<T>& data)
+    {
+        std::vector<std::size_t> indices;
+        for(std::size_t i = 0; i < data.size(); ++i)
+        {
+            if(data[i] != 0.0)
```
Comment (kahmed10) [inline src/onnx/onnx.cpp:2331]: I think we can use `float_equal` here

Comment (scxiao) [reply src/onnx/onnx.cpp]: done

File: src/onnx/onnx.cpp [hunk 3]
```diff
@@ -2320,6 +2323,49 @@ struct onnx_parser
         MIGRAPHX_THROW("PARSE_ATEN: unsupported custom operator");
     }
 
+    template <class T>
+    std::vector<std::size_t> nonzero_indices(std::vector<T>& data)
```
Comment (pfultz2) [inline src/onnx/onnx.cpp:2327]: Should `data` be `const`?

Comment (scxiao) [reply src/onnx/onnx.cpp]: fixed

**Comment by scxiao** [comment]
@kahmed10 if you have no additional comments, can you approve it?

**Pull request merged**